src/generate-query.ts

```diff
@@ -610,6 +669,145 @@ function getArgsAndVars(
   }
 }
 
+function getListSizeDirective(field: FieldDefinitionNode): DirectiveNode {
+  return field?.directives.find((dir) => dir.name.value === 'listSize')
+}
+
+function getSlicingArguments(
+  field: FieldDefinitionNode,
+  listSizeDirective: DirectiveNode
+): {
+  slicingArgumentsArg: ArgumentNode
+  listSizeDirective: DirectiveNode
+} {
+  listSizeDirective = listSizeDirective
+    ? listSizeDirective
+    : getListSizeDirective(field)
+  if (!listSizeDirective)
+    return { slicingArgumentsArg: undefined, listSizeDirective: undefined }
+
+  const slicingArgumentsArg = listSizeDirective.arguments.find(
+    (arg) => arg.name.value === 'slicingArguments'
+  )
+
+  if (slicingArgumentsArg && slicingArgumentsArg.value.kind !== 'ListValue')
+    return { slicingArgumentsArg: undefined, listSizeDirective: undefined }
+
+  return { slicingArgumentsArg, listSizeDirective }
+}
+
+function getRequireOneSlicingArgument(
```

Comment:
I have similar thoughts with this function. I think it should just be`getRequireOneSlicingArgumentFromListSizeDirective()` or better yet, is there a way to combine both this and `getSlicingArguments()` so it just extracts all the relevant information from a list size directive?

Reply:
We could combine them, but I might like the piecemeal approach more. I'll switch it to a simpler function then.